mlflow/pyspark/ml/__init__.py

```diff
@@ -1098,6 +1098,9 @@ def _infer_model_signature(input_example_slice):
                         )
                         log_model_signatures = False
 
+                # `_infer_spark_model_signature` mutates the model. Copy the model to preserve the
+                # original model.
+                spark_model = spark_model.copy()
```

Comment:
Do we need to only do this for actual model objects? The ephemeral objects within Cross Validators and Estimators won't have the full attributes of an actual model object, right?